### PR TITLE
create order history page

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: src/components/OrderHistoryItem.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      </q-item-main>
+      <q-item-side class="text-weight-bold">{{'$'+rawData.totalAmount}}</q-item-side>
+    </q-item>
+  </div>
+</template>
+<script>
+import {date} from 'quasar'
+export default {
+  props: {
+    rawData: {
```
Comment (hemvanh) [inline src/components/OrderHistoryItem.vue:18]: rawData -> should initialize default values for rawData

File: src/components/OrderHistoryItem.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <div class="order-item">
+    <q-item>
+      <q-item-side :icon="rawData.isStorePickUp?'store':'local_shipping'"/>
+      <q-item-main>
+        <q-item-tile label class="text-weight-bold" text-color="primary">{{rawData.address}}</q-item-tile>
+        <q-item-tile sublabel>{{rawData.statusName}}</q-item-tile>
+        <q-item-tile sublabel text-color="secondary">{{rawData.orderDateTime|formatedDate}}</q-item-tile>
```
Comment (hemvanh) [inline src/components/OrderHistoryItem.vue]: orderDateTime -> orderDate
formatedDate -> formatOrderDate
statusName -> orderStatus

File: src/store/customer-order/state.js [hunk 3]
```diff
@@ -10,4 +10,5 @@ export default {
     pickUpTime: new Date().setMinutes(new Date().getMinutes() + 30),
     isStorePickUp: false,
   },
+  customerOrder: [],
```
Comment (hemvanh) [inline src/store/customer-order/state.js]: customerOrders

File: src/store/customer-order/getters.js [hunk 4]
```diff
@@ -7,3 +7,6 @@ export const getIsLoading = state => {
 export const getPlaceOrderMethod = state => {
   return state.placeOrderMethod
 }
+export const getCustomerOrder = state => {
```
Comment (hemvanh) [inline src/store/customer-order/getters.js]: getCustomerOrders

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/OrderHistoryItem.vue (hunk 1 above)
Comment (tri-et) [reply src/components/OrderHistoryItem.vue:18]: u mean set default values for rawData like this?
 props: {
    rawData: {
      type: Object,
      default: () => ({
        address: '',
        orderDate: '',
        orderStatus: '',
        totalAmount: 0,
        isStorePickUp: false,
      }),
    },
  },

Comment (hemvanh) [reply src/components/OrderHistoryItem.vue:18]: yes, this is for self-documenting -> when other programmer look at the vue component file, they should know what to pass to rawData

File: src/store/customer-order/actions.js [hunk 5]
```diff
@@ -26,3 +26,24 @@ export const placeOrder = ({commit, getters}) => {
       })
   } else _alert('Please login first!', 'warning')
 }
+export const fetchCustomerOrder = ({commit}) => {
```
Comment (hemvanh) [inline src/store/customer-order/actions.js]: fetchCustomerOrder -> fetchCustomerOrders

**Review by hemvanh — CHANGES_REQUESTED** [review]

**Pull request merged**